source/extensions/omni.isaac.lab/omni/isaac/lab/assets/rigid_object/rigid_object_data.py

```diff
@@ -121,20 +129,26 @@ def root_state_w(self):
     def body_state_w(self):
         """State of all bodies `[pos, quat, lin_vel, ang_vel]` in simulation world frame. Shape is (num_instances, 1, 13).
 
-        The position and orientation are of the rigid bodies' actor frame. Meanwhile, the linear and angular
-        velocities are of the rigid bodies' center of mass frame.
+        All values are relative to the world.
         """
         return self.root_state_w.view(-1, 1, 13)
 
     @property
     def body_acc_w(self):
         """Acceleration of all bodies. Shape is (num_instances, 1, 6).
 
-        This quantity is the acceleration of the rigid bodies' center of mass frame.
+        All values are relative to the world.
         """
         if self._body_acc_w.timestamp < self._sim_timestamp:
             # note: we use finite differencing to compute acceleration
             self._body_acc_w.data = self._root_physx_view.get_accelerations().unsqueeze(1)
+            # move linear acceleration to link frame
+            ang_acc_w = self._body_acc_w.data[..., 3:]
+            ang_vel_w = self.body_state_w[..., 7:10]
+            com_pos_w = math_utils.quat_rotate(self.body_state_w[..., 3:7], -self._com_pos_b.unsqueeze(1))
+            self._body_acc_w.data[..., :3] += torch.linalg.cross(ang_acc_w, com_pos_w, dim=-1) + torch.linalg.cross(
+                ang_vel_w, torch.linalg.cross(ang_vel_w, com_pos_w, dim=-1), dim=-1
+            )
```

Comment:
I can break them out into util.math functions but they will have to be separate from the Articulation and RigidObject classes because there isn't inheritance any more.